pkg/flare/metadata.go

```diff
@@ -0,0 +1,51 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package flare
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/inventories"
+	v5 "github.com/DataDog/datadog-agent/pkg/metadata/v5"
+	"github.com/DataDog/datadog-agent/pkg/util"
+)
+
+func addMetadata(tempDir, hostname, filename string, data []byte) error {
+	f := filepath.Join(tempDir, hostname, filename)
+	w, err := NewRedactingWriter(f, os.ModePerm, true)
```

Comment:
This risks causing cause failures on `main` if not merged carefully with https://github.com/DataDog/datadog-agent/pull/9682.

Reply:
#9682 has landed, and I merged from master.  I caught this issue in the merge.